### PR TITLE
Implement `PropertyPath.Add`.

### PR DESCRIPTION
**Pull request by pgavlin**
This function adds a property value to another property at a given path,
creating containing properties as required. If the property cannot be
added because of a mismatch between the value types required by the path
and the values present in the destination, the add will fail. If a value
already exists at the given path, the add will succeed.

### CODE REVIEW

**Comment by pgavlin** [comment]
Related to #1635.

File: sdk/go/common/resource/properties_path.go [hunk 1]
```diff
@@ -160,6 +160,63 @@ func (p PropertyPath) Set(dest, v PropertyValue) bool {
...
+	for _, key := range p {
+		switch key := key.(type) {
+		case int:
+			switch {
+			case dest.IsNull():
+				dest = NewArrayProperty(make([]PropertyValue, key+1))
+				set(dest)
+			case dest.IsArray():
+				arr := dest.ArrayValue()
+				if key >= len(arr) {
```
Comment (lblackstone) [inline sdk/go/common/resource/properties_path.go:184]: I don't understand why the value is being compared to the length of the array.

Comment (pgavlin) [reply sdk/go/common/resource/properties_path.go:191]: Added some comments to clarify.

Comment (EvanBoyle) [reply sdk/go/common/resource/properties_path.go:191]: The key in this instance is the array index, and this code checks for space in the array to accommodate the element?

**Pull request merged**